### ISSUE TITLE
Integers bound usage

### ISSUE THREAD

**Issue by anaelChardan** [Innmind/BlackBox#3]
Hi!

Thanks a lot for your lib! So useful and easy to use.
I encounter a problem with Integers Bound.

I had to add a filter to respect the bounds in the `between` part

```
        return Decorate::mutable(
            fn (int $number): BrandCreationYear => BrandCreationYear::belowCurrentYear($number),
            Set\Integers::between(1800, ((int) date('Y')))
                ->filter(fn (int $value) => $value > 1800 && $value <= (int) date('Y'))
        );
```

At a first sight it looks like it could be due to the shrink method that is doing a dichotomy ;)

Thank you in advance!

**Comment by Baptouuuu** [comment]
Hi,

It stems from the `random_int` function using inclusive bounds, I noticed this last month that it was incoherent between what the constructors suggest and the actual behaviour.

I didn't change the behaviour at the time out of fear of introducing a bc break. Would you be ok if I had new named constructors like `betweenExclusiveBounds($a, $b)` ?

On another note I will change the behaviour of `above` and `below` as the name can't suggest the value being present in the resulting set.

**Comment by Baptouuuu** [comment]
As discussed off issue, the problem lies in the fact that shrunk values do not respect initial bounds . (knowing this my previous comment can be discarded)

The problem is also present in `RealNumbers` and `Strings` sets.

A fix has been released with the version number `4.2.1`.

I'll let you close this issue when you'll be able to confirm the problem is indeed fixed on your side.

**Comment by anaelChardan** [comment]
Hi it works ! thanks a lot!
